hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinySessionFactoryImpl.java

```diff
@@ -151,33 +164,36 @@ private ReactiveConnection proxyConnection(String tenantId) {
 
 	@Override
 	public <T> Uni<T> withStatelessSession(Function<Mutiny.StatelessSession, Uni<T>> work) {
-		Mutiny.StatelessSession current = context.get( Mutiny.StatelessSession.class, statelessSessionId );
+		Objects.requireNonNull( work, "parameter 'work' is required" );
+		Mutiny.StatelessSession current = context.get( Mutiny.StatelessSession.class, contextKeyForStatelessSession );
 		if ( current != null && current.isOpen() ) {
 			return work.apply( current );
 		}
-		return withSession( Mutiny.StatelessSession.class, newStatelessSession(), work, statelessSessionId );
+		return withSession( Mutiny.StatelessSession.class, newStatelessSession(), work, contextKeyForStatelessSession );
 	}
 
 	private<S extends Mutiny.Closeable, T> Uni<T> withSession(
-			Class<S> sessionType,
-			Uni<S> sessionUni,
-			Function<S, Uni<T>> work,
-			String id) {
+			final Class<S> sessionType,
+			final Uni<S> sessionUni,
+			final Function<S, Uni<T>> work,
+			final String contextId) {
```

Comment:
Why id this necessary? Surely these are marked implicitly-final by the compiler.

Reply:
I agree, they are most likely unnecessary. Personally I prefer such things to be explicit - matter of style and taste :)

Can revert if you don't like it, no strong opinion.

Reply:
Well we don't do that anywhere else in the codebase: we rely on implicit final _all over the place_. It's weird to not trust it in just one method.

Reply:
Removed it.

But I generally never trust it and when I code I prefer it explicit . Especially when we have many lambdas, there's a high degree of risk of unintentionally creating a performance problem as it's very easy to miss cases in which the code is no longer eligible for "effectively final" treatment - and when it's common practice to not use them it's kinda guaranteed we'll have issues with it eventually.